kube/src/api/resource.rs

```diff
@@ -1,118 +1,83 @@
-            kind: K::kind(dyntype).into_owned(),
-            group: K::group(dyntype).into_owned(),
-            version: K::version(dyntype).into_owned(),
+            info,
             namespace: None,
         }
     }
 
     /// Namespaced resource within a given namespace
     ///
-    /// This function accepts `K::DynamicType` so it can be used with dynamic resources.
-    pub fn namespaced_with<K: Meta>(ns: &str, dyntype: &K::DynamicType) -> Self {
-        let kind = K::kind(dyntype);
-        match kind.as_ref() {
-            "Node" | "Namespace" | "ClusterRole" | "CustomResourceDefinition" => {
-                panic!("{} is not a namespace scoped resource", kind)
-            }
-            _ => {}
-        }
+    /// This function accepts `K::Info` so it can be used with dynamic resources.
+    pub fn namespaced_with(ns: &str, info: K::Info) -> Self {
         Self {
-            api_version: K::api_version(dyntype).into_owned(),
-            kind: kind.into_owned(),
-            group: K::group(dyntype).into_owned(),
-            version: K::version(dyntype).into_owned(),
+            info,
             namespace: Some(ns.to_string()),
         }
     }
 
+    // TODO: provide namespace setter taking mut self
+
     /// Cluster level resources, or resources viewed across all namespaces
-    pub fn all<K: Meta>() -> Self
+    pub fn all() -> Self
     where
-        <K as Meta>::DynamicType: Default,
+        <K as Meta>::Info: Default,
     {
-        Self::all_with::<K>(&Default::default())
+        Self::all_with(Default::default())
     }
 
     /// Namespaced resource within a given namespace
-    pub fn namespaced<K: Meta>(ns: &str) -> Self
+    pub fn namespaced(ns: &str) -> Self
     where
-        <K as Meta>::DynamicType: Default,
+        <K as Meta>::Info: Default,
     {
-        Self::namespaced_with::<K>(ns, &Default::default())
+        Self::namespaced_with(ns, Default::default())
     }
 
-    /// Manually configured resource or custom resource
-    ///
-    /// This is the only entrypoint to `Resource` that bypasses [`k8s_openapi`] entirely.
-    /// If you need a `CustomResource` consider using `kube-derive` for its
-    /// `#[derive(CustomResource)]` proc-macro.
-    pub fn dynamic(kind: &str) -> DynamicResource {
-        DynamicResource::new(kind)
-    }
+    // Build a custom Request type from a `GroupVersionKind`
+    //pub fn dynamic(kind: &str) -> RequestBuilder {
+    //    RequestBuilder::new(kind)
+    //}
 }
 
 // -------------------------------------------------------
 
-impl Resource {
+impl<K: Meta> Request<K> {
     pub(crate) fn make_url(&self) -> String {
         let n = if let Some(ns) = &self.namespace {
             format!("namespaces/{}/", ns)
         } else {
             "".into()
         };
+        let group = K::group(&self.info);
+        let api_version = K::api_version(&self.info);
+        let plural = to_plural(&K::kind(&self.info).to_ascii_lowercase());
```

Comment:
Rather than storing the resource data in this struct temporarily, we just grab it out of the trait (whose associated type we have at hand).